app/javascript/dashboard/components/ChatList.vue

```diff
@@ -364,13 +370,37 @@ export default {
     },
     allConversationsSelected() {
       return (
-        JSON.stringify(this.selectedConversations) ===
-        JSON.stringify(this.conversationList.map(item => item.id))
+        this.conversationList.length === this.selectedConversations.length &&
+        this.conversationList.every(el =>
+          this.selectedConversations.includes(el.id)
+        )
       );
     },
     uniqueInboxes() {
       return [...new Set(this.selectedInboxes)];
     },
+    allSelectedConversationsAreOpen() {
+      if (!this.selectedConversations.length) return false;
+      return this.selectedConversations.every(item => {
+        return this.$store.getters.getConversationById(item).status === 'open';
+      });
+    },
+    allSelectedConversationsAreResolved() {
+      if (!this.selectedConversations.length) return false;
+      return this.selectedConversations.every(item => {
+        return (
+          this.$store.getters.getConversationById(item).status === 'resolved'
+        );
+      });
```

Comment:
This logic is repeated. Why don't we just pass the status as a param and make this a function?